src/Internal/Payment/Factor.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class Factor
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+/**
+ * A class for payment factors.
+ *
+ * These factors will be used to determine whether to enter the new
+ * payment process until it is fully developed and this class gets removed.
+ */
+class Factor {
```

Comment:
@RadoslavGeorgiev  

1. Agreed that his is annoying. I am not certain how to avoid this. May remove the `@throw` tag in the `Base_Constant::__callStatic` method, but not ideal anyway. 
2. I looked at `Base_Constant` implementation, and it appears that we would improve it so that each `__callStatic()` would not create a new object. Instead, we cache initialized objects in a static property, and return the same object for the same static call function, which is similar to the singleton pattern. Something like this https://github.com/Automattic/woocommerce-payments/pull/7063 (and more in this commit  https://github.com/Automattic/woocommerce-payments/pull/7063/commits/ae6f3318589d840dc4ed48228a7d70ccaac45a06)
3. IMO, this is acceptable for array keys (indexes).

Reply:
I like the idea there, especially the fact that we can use `===` to compare instances because there would not be more than one instance of the same constant. I think it's worth proceeding with that PR.

Reply:
Moving that one to a new PR https://github.com/Automattic/woocommerce-payments/pull/7064

Reply:
https://github.com/Automattic/woocommerce-payments/pull/7064 is approved, I'll implement the changes here once it gets merged.